git-traverse/src/commit.rs

```diff
@@ -109,6 +167,63 @@ pub mod ancestors {
             res.map(Ok)
         }
     }
+
+    mod predicates {
+        use super::*;
+
+        /// `FnMut`-like trait used for pruning commit subgraphs from the ancestors iterator.
+        ///
+        /// This trait only exists because:
+        /// 1. Predicate usage should be optional.
+        ///    The user should opt-in to using predicates via an alternate constructor or via the
+        ///    builder pattern.
+        /// 2. The default case (include all ancestors) should not increase the size of the
+        ///    [`Ancestors`] struct, nor should it require extra branches at runtime.
```

Comment:
I agree with the second point, but is there significant harm in increasing the size? The difference should be at most a `usize`, in the case of using a function pointer, right? Are there regressions with a size increase?

Reply:
Correct, if we can live with a `usize` increase, then we can use `impl<...> Ancestors <Find, fn(&oid) -> bool, StateMut>` and avoid all of the `OidPredicate` stuff. I'd also need to check whether that approach destroys the compiler's ability to inline the default predicate.